airflow/cli/utils/console.py

```diff
@@ -0,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Console used by all processes. We are forcing colors and terminal output as Breeze is supposed
+to be only run in CI or real development terminal - in both cases we want to have colors on.
+"""
+try:
+    from rich.console import Console
+
+    console = Console()
+
+except ImportError:
+    custom_theme = None  # type: ignore[assignment]
+    console = None  # type: ignore[assignment]
```

Comment:
We could import console from this file rather than creating an instance every time in each function where we need to print. I thought it would be better to have console and its formatting in seperate file. 
https://github.com/astronomer/airflow/blob/8e87a0766577bea08f6a4d2c9276540db6b4584e/airflow/cli/commands/scheduler.py#L96-L100

Reply:
```python
 def scheduler(ctx, daemon_, do_pickle, log_file, num_runs, pid, skip_serve_logs, stderr, stdout, subdir): 
     """Starts Airflow Scheduler""" 
     console = Console() 
     console.print(settings.HEADER) 
```

That code ^ is only executed when the command/function is called, not when it is imported.

This code:

```python
try:
    from rich.console import Console

    console = Console()
```

immediately imports and executes `console = Console()` when the module is imported. And you generally only want to define things at import time (eg: `class ...`, `def ...`), you don't want to do any "work" or actually _call_ any functions (eg: `console = Console()`), because those can [start having side effects](https://chrismorgan.info/blog/say-no-to-import-side-effects-in-python/).

Python caches imports, so after the first time it imports `rich.console` it shouldn't appreciably impact the startup time of the commands. However, if it makes a big difference, we can move the `from rich.console import Console` line into each function. If we do that, we will still pay the time to import and call `Console()`, but Python/Click won't do any of that for users who are just running the various help commands: `airflow-ng --help`, `airflow-ng db --help`, `airflow-ng db upgrade --help`, etc.

If I'm not understanding something more subtle here please let me know.

Also, is `airflow.cli.utils.console.custom_theme` imported or used anywhere? I couldn't find anywhere it's used, so I'm not sure what this code:

```python
except ImportError:
    custom_theme = None
```

is trying to do.

Reply:
@blag we have used something similar in breeze2 development work for print console. I thought it would be relevant to add here and added that part of the code here. we have added custom theme to set uniform theme of `INFO, WARNING, ERROR`.  If we don't have such need here, I will remove this part of code

Reply:
Sorry, I'm not understanding where `airflow.cli.utils.console.custom_theme` is accessed. Can you explain what code uses that variable?

Reply:
@blag we have not used `airflow.cli.utils.console.custom_theme` anywhere till now. https://github.com/apache/airflow/blob/07a13bb7088d4a8ba50efbf14981150656b020a7/dev/breeze/src/airflow_breeze/utils/console.py#L17-L42

Similar code is used in breeze2 and I thought it would be useful here and added that part.

Reply:
I still don't see where we're using this `custom_theme` anywhere. It looks like we're only defining it here but not using it. Can you explain how and where it's used?

Reply:
i am removing this part @blag as its not used anywhere now.